translator/translate/otel/extension/ecsobserver/translator.go

```diff
@@ -0,0 +1,183 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package ecsobserver
+
+import (
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/ecsobserver"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
+)
+
+const (
+	defaultMetricsPath      = "/metrics"
+	defaultPortLabel        = "ECS_PROMETHEUS_EXPORTER_PORT"
+	defaultMetricsPathLabel = "ECS_PROMETHEUS_METRICS_PATH"
+	defaultJobNameLabel     = "job"
+)
+
+var ecsSDKey = common.ConfigKey(common.LogsKey, common.MetricsCollectedKey, common.PrometheusKey, common.ECSServiceDiscovery)
+
+type translator struct {
+	factory component.Factory
+	name    string
+}
+
+var _ common.ComponentTranslator = (*translator)(nil)
+var _ common.NameSetter = (*translator)(nil)
+
+func NewTranslator(opts ...common.TranslatorOption) common.ComponentTranslator {
+	t := &translator{
+		factory: ecsobserver.NewFactory(),
+	}
+	for _, opt := range opts {
+		opt(t)
+	}
+	return t
+}
+
+func (t *translator) ID() component.ID {
+	return component.NewIDWithName(t.factory.Type(), t.name)
+}
+
+func (t *translator) SetName(name string) {
+	t.name = name
+}
+
+func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
+	if conf == nil || !conf.IsSet(ecsSDKey) {
+		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: ecsSDKey}
+	}
+
+	ecsSD := common.GetIndexedMap(conf, ecsSDKey, -1)
```

Comment:
Why are we using `GetIndexedMap`? This is typically used to get an element out of an array. If it's not an array, I would suggest sticking with the confmap.